scripts/deploy-multidev.sh

```diff
@@ -1,27 +1,81 @@
 #!/bin/bash
-# Note: Clear cache on $ENV after executing this script
-
-# Define env and site UUID
-export ENV=docs-files
-export SITE=72e163bd-0054-4332-8bf8-219c50b78581
-# Create local directory and file to log rsync output (local path: "$HOME/sites/docs-backups")
-mkdir ../docs-backups
-mkdir ../docs-backups/`date +%F-%I%p`
-echo "rsync log - deploy to docsfiles Multidev environment on `date +%F-%I%p`" > ../docs-backups/`date +%F-%I%p`/rsync-`date +%F-%I%p`.log
-# Build prod
-rm -rf output_prod/docs
-sculpin generate --env=prod
-cd output_prod
-
-### Get path for deploy from user and deploy only specified doc
-read -p "What doc or directory? (e.g. docs/articles/users/index.html or docs/articles/users/): " path
-[ -z "${path}" ] && path='docs'
-if [ ! -e $path ]
+# Deploy branch to multidev on commit
+# Note: This script uses CircleCI environment variables https://circleci.com/docs/environment-variables
+# Note: PRs from forks not yet supported, see: https://circleci.com/docs/fork-pr-builds
+
+# Deploy any branch except master, dev, test, or live
+if [ "$CIRCLE_BRANCH" != "master" ] && [ "$CIRCLE_BRANCH" != "dev" ] && [ "$CIRCLE_BRANCH" != "test" ] && [ "$CIRCLE_BRANCH" != "live" ]; then
+
+    # Normalize branch name to adhere with Multidev requirements
+    export normalize_branch="$CIRCLE_BRANCH"
+    export valid="^[-0-9a-z]" # allows digits 0-9, lower case a-z, and -
+    if [[ $normalize_branch =~ $valid ]]; then
+        export normalize_branch="${normalize_branch:0:11}"
+        echo "Success: "$normalize_branch" is a valid branch name."
+    else
+        echo "Error: Multidev cannot be created due to invalid branch name: $normalize_branch"
```

Comment:
@bmackinney The error statement occurs when characters other than those allowed are present. So it would fail on `1234_php_log_update` and stop the build but `1234-php-log-update` truncates to `1234-php-lo` and the build continues.